### ISSUE TITLE
[NEXTCLOUD] Can not set custom OVERWRITEHOST

### ISSUE THREAD

**Issue by b-m-f** [truenas/charts#1151]
Because the chart logic is being executed after user-provided content is being parsed, everything I set as a user is being overwritten.

Since I need to have control over those settings in my deployment it would be nice to have over provided `ENV` variables be included after the internal chart logic.

Problem code: https://github.com/truenas/charts/blob/386e6770c71db99b7f009399739ccf7109067700/charts/nextcloud/1.6.22/templates/deployment.yaml#L82

**Comment by stavros-k** [comment]
If you select a certificate, you can set the `Nextcloud Host` and `Node Port to use for Nextcloud` Which will be combined to set the `OVERWRITEHOST`.

Without certificate you can add additiional environment variables and set it yourself.

What are you trying to do exactly and this blocking you?
Thanks

**Comment by b-m-f** [comment]
My problem is the following:

I set the environment variable `OVERWRITEHOST` to `nextcloud.domain` via the settings.

Now when I start the app the url still defaults to `truenas.domain:Port`.

I believe that this is because
```
        {{ $envList := (default list .Values.environmentVariables) }} <---------------------------
        {{ $envList = mustAppend $envList (dict "name" "POSTGRES_HOST" "value" (printf "%s:5432" (include "common.names.fullname" $postgres_values))) }}
        {{ $envList = mustAppend $envList (dict "name" "POSTGRES_DB" "value" (include "postgres.DatabaseName" .)) }}
        {{ $envList = mustAppend $envList (dict "name" "NEXTCLOUD_DATA_DIR" "value" .Values.nextcloud.datadir) }}
        {{ if eq (include "nginx.certAvailable" .) "true" }}
        {{ $envList = mustAppend $envList (dict "name" "APACHE_DISABLE_REWRITE_IP" "value" "1") }}
        {{ $envList = mustAppend $envList (dict "name" "OVERWRITEHOST" "value" (printf "%v:%v" .Values.nextcloud.host .Values.service.nodePort)) }} <---------------------------
```
`{{ $envList := (default list .Values.environmentVariables) }}` should be set further down so my user provided values will overwrite the default.

> What are you trying to do exactly and this blocking you?

I need the host to be configured to the value that my proxy will use.
With the current behaviour I am blocked on the defaults and need to manually edit nextclouds `.php` files in order to stop the wrong Host from being set.
